starter_code/app.js

```diff
@@ -1,26 +1,75 @@
 const express = require('express');
 const hbs = require('hbs');
-
-// require spotify-web-api-node package here:
-
-
-
+const SpotifyWebApi = require('spotify-web-api-node');
 const app = express();
 
 app.set('view engine', 'hbs');
 app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
-
-
+hbs.registerPartials(__dirname + '/views/partials');
 // setting the spotify-api goes here:
-
-
-
-
-
+const clientId = '06d72fb9a7bb4f15a42dfe168c6e950f',
+  clientSecret = '8edebf9b17d0466eaa65735a72361391';
+
+const spotifyApi = new SpotifyWebApi({
+  clientId: clientId,
+  clientSecret: clientSecret
+});
+//token
+spotifyApi
+  .clientCredentialsGrant()
+  .then(data => {
+    spotifyApi.setAccessToken(data.body['access_token']);
+  })
+  .catch(error => {
+    console.log('Something went wrong when retrieving an access token', error);
+  });
 
 // the routes go here:
-
-
-
-app.listen(3000, () => console.log("My Spotify project running on port 3000 🎧 🥁 🎸 🔊"));
+app.get('/', (req, res) => {
+  res.render('home.hbs');
+});
+
+app.get('/artists', (req, res) => {
+  spotifyApi
```

Comment:
you can see sometimes another way to do the query by deconstructing here:
`const { artistName } = request.query;`
then you can only apply artistName as an argument